lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -531,7 +538,18 @@ def _run_script(self, rerun_data: RerunData) -> None:
 
                 ctx.on_script_start()
                 prep_time = timer() - start_time
-                exec(code, module.__dict__)
+                if rerun_data.partial_id:
+                    partial_func = load_partial(rerun_data.partial_id)
+                    if not partial_func:
+                        raise RuntimeError(
+                            f"Could not find partial with id {rerun_data.partial_id}"
+                        )
+                    partial_func()
+                    partial_run = True
+                else:
+                    # Clean partial storage:
+                    PartialsStorage().clear()
```

Comment:
Every rerun cleans all partials since a full rerun will generate the partials again anyway. This is to make sure that we don't have any memory leaks. This might not work if we store partials in a way that is not bound to a session.